pkg/balancer/least_connections.go

```diff
@@ -0,0 +1,128 @@
+package balancer
+
+import (
+	"math"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/square/p2/pkg/util"
+)
+
+type lcBackend struct {
+	conns    int
+	disabled bool
+}
+
+type LeastConnections struct {
```

Comment:
yeah there will always be potential for a bad host that you can't detect. Health check might work but a typo in a rails project might cause requests to fail for example